src/analytics/navigationTracker/navigationTracker.test.ts

```diff
@@ -0,0 +1,171 @@
+    const href = document.createElement("div");
+    href.className = "govuk-footer__link";
+    href.addEventListener("click", (event) => {
+      expect(newInstance.trackNavigation(event)).toBe(false);
+    });
+    href.dispatchEvent(action);
+  });
+
+  //test trackNavigation accept button
+  test("trackNavigation should return true if a link", () => {
+    const href = document.createElement("A");
+    href.className = "govuk-footer__link";
+    href.innerHTML = "Link to GOV.UK";
+    href.setAttribute("href", "http://localhost");
+    href.addEventListener("click", (event) => {
+      expect(newInstance.trackNavigation(event)).toBe(true);
+    });
+    href.dispatchEvent(action);
+  });
+
+  //test trackNavigation accept link
+  test("trackNavigation should return true if a button", () => {
+    const href = document.createElement("BUTTON");
+    href.className = "govuk-footer__link";
+    href.innerHTML = "Link to GOV.UK";
+    href.setAttribute("href", "http://localhost");
+    href.addEventListener("click", (event) => {
+      expect(newInstance.trackNavigation(event)).toBe(true);
+    });
+    href.dispatchEvent(action);
+  });
+
+  //test pushToDataLayer is called
+  test("pushToDataLayer is called", () => {
+    const href = document.createElement("A");
+    href.className = "govuk-footer__link";
+    href.addEventListener("click", (event) => {
+      newInstance.trackNavigation(event);
+    });
+    href.dispatchEvent(action);
+    expect(newInstance.pushToDataLayer).toBeCalled();
+  });
+});
+
+describe("getLinkType", () => {
+  const newInstance = new NavigationTracker();
+  const action = new MouseEvent("click", {
+    view: window,
+    bubbles: true,
+    cancelable: true,
+  });
+
+  test('should return "footer" when the element is an <a> tag with footer link class', () => {
+    const href = document.createElement("A");
+    href.className = "govuk-footer__link";
+    href.dispatchEvent(action);
+    const element = action.target as HTMLLinkElement;
+    expect(newInstance.getLinkType(element)).toBe("footer");
+  });
+
+  test('should return "header menu bar" when the element is an <a> tag with header menu bar link class', () => {
+    const href = document.createElement("A");
+    href.className = "header__navigation";
+    href.dispatchEvent(action);
+    const element = action.target as HTMLLinkElement;
+    expect(newInstance.getLinkType(element)).toBe("header menu bar");
+  });
+
+  test('should return "generic link" when the element is an <a> tag without footer or header menu bar link class', () => {
+    const href = document.createElement("A");
+    href.className = "other-link";
+    href.dispatchEvent(action);
+    const element = action.target as HTMLLinkElement;
+    expect(newInstance.getLinkType(element)).toBe("generic link");
+  });
+
+  test('should return "generic button" when the element is a <button> tag', () => {
+    const button = document.createElement("BUTTON");
```

Comment:
Good question. I use caps because during my test, when I receive an event from the click event listener (line 20 navigationTracker.js), the tag was always CAPS. So in the test, I did the same. So technically it's just because I want some consistency between unit test and irl browser interaction.